### ISSUE TITLE
Default if value is empty

### ISSUE THREAD

**Issue by MadLittleMods** [postcss/postcss-mixins#33]
When nesting mixins, passing along variable that is not defined, should also be undefined and default in the nested mixin.

Currently it seems to fill it with a blank value which makes it not use the default fallback value.

### [Demo](https://tonicdev.com/mlm/564217b66ca5f20c0078bc3f)

```css
@define-mixin style-links $color: #f00 {
    foo: $color;
}

@define-mixin format-content $color {
    @mixin style-links $color;
}

.foo {
    @mixin format-content;
}

.bar {
    @mixin format-content #0ff;
}
```

I would expect:

```css
.foo {
    foo: #f00;
}

.bar {
    foo: #0ff;
}
```

But currently the output is:

```css
.foo {
    foo: ;
}

.bar {
    foo: #0ff;
}
```

**Comment by ai** [comment]
It do not supported inn current architecture. You should ask @jonathantneal to add more magic in PreCSS or wait for Event Based API.

**Comment by MadLittleMods** [comment]
@ai nested mixins work just fine, it seems the variable resolution just needs some tweaking.

A workaround for now is just to define the defaults on everything:

```
@define-mixin style-links $color: #f00 {
    foo: $color;
};

@define-mixin format-content $color: #f00 {
    @mixin style-links $color;
};

.foo {
    @mixin format-content;
}

.bar {
    @mixin format-content #0ff;
}
```

**Comment by ai** [comment]
Fixed 2f8f0c55f542bca5d00439cdee586d417fb79ad1

**Comment by ai** [comment]
Released in 3.0.1
